keychain/src/utils/account.ts

```diff
@@ -42,12 +44,38 @@ class Account extends BaseAccount {
     });
 
     try {
-      const classHash = await this.rpc.getClassHashAt(this.address, "latest");
-      Storage.update(this.selector, {
-        classHash,
-        deployed: true,
-      });
-      this.deployed = true;
+      const state = Storage.get(this.selector);
+      if (!state.deployed) {
+        const deployTx = Storage.get(this.selector).deployTx;
+        if (deployTx) {
+          const receipt = (await this.rpc.getTransactionReceipt(
+            deployTx,
+          )) as GetTransactionReceiptResponse;
+          switch (receipt.status) {
+            case "ACCEPTED_ON_L1":
+            case "ACCEPTED_ON_L2":
+              this.deploying = false;
+              this.deployed = true;
+              break;
+            case "RECEIVED":
+            case "PENDING":
+              this.deploying = true;
+              this.deployed = false;
+              throw new Error("Deploying");
```

Comment:
can we return here and below instead of throwing?

Reply:
Yep, just turned it into a method to keep it sync cleaner and easier to call from frontend too I think